File: app/services/modlog.py
"""Send moderation events to admin log channel + SQLite."""
from __future__ import annotations

import logging

from aiogram import Bot

from app.config import Settings
from app.database import log_action

log = logging.getLogger(__name__)


async def record(
    bot: Bot,
    settings: Settings,
    *,
    chat_id: int,
    user_id: int,
    action: str,
    reason: str = "",
    by_user: int = 0,
    chat_title: str = "",
) -> None:
    await log_action(settings.db_path, chat_id, user_id, action, reason, by_user)
    channel = settings.log_channel_id
    if not channel:
        return
    text = (
        f"🛡 <b>{action}</b>\n"
        f"Group: {chat_title or chat_id} (<code>{chat_id}</code>)\n"
        f"User: <code>{user_id}</code>\n"
        f"By: <code>{by_user}</code>\n"
        f"Reason: {reason or '-'}"
    )
    try:
        await bot.send_message(channel, text)
    except Exception as e:  # noqa: BLE001 - log channel must never crash moderation
        log.warning("Failed to write modlog to %s: %s", channel, e)

File: node-bot/lib/modlog.js
'use strict';
const store = require('./store');

function cfg() {
  return {
    logChannel: (process.env.LOG_CHANNEL_ID || '').trim(),
  };
}

async function record(bot, { chatId, userId, action, reason = '', byUser = 0, chatTitle = '' }) {
  store.logAction(chatId, userId, action, reason, byUser);
  const { logChannel } = cfg();
  if (!logChannel) return;
  try {
    await bot.api.sendMessage(
      logChannel,
      `🛡 <b>${action}</b>\nGroup: ${chatTitle || chatId} (<code>${chatId}</code>)\nUser: <code>${userId}</code>\nBy: <code>${byUser}</code>\nReason: ${reason || '-'}`,
      { parse_mode: 'HTML' }
    );
  } catch (e) {
    console.warn('modlog send failed:', e.message);
  }
}

module.exports = { record };
